Sources/SwiftPlot/GraphLayout.swift

```diff
@@ -71,127 +82,220 @@ public struct GraphLayout {
-        if let xLabelSize = results.sizes.xLabelSize {
-            results.xLabelLocation = Point(results.plotBorderRect.midX - xLabelSize.width/2,
-                                           Self.xLabelPadding)
-        }
-        if let titleLabelSize = results.sizes.titleSize {
-            results.titleLocation = Point(results.plotBorderRect.midX - titleLabelSize.width/2,
-                                          results.totalSize.height  - titleLabelSize.height - Self.titleLabelPadding)
-        }
-        if let yLabelSize = results.sizes.yLabelSize {
-            results.yLabelLocation = Point(Self.yLabelPadding + yLabelSize.height,
-                                           results.plotBorderRect.midY - yLabelSize.width/2)
-        }
-        if let y2LabelSize = results.sizes.y2LabelSize {
-            results.y2LabelLocation = Point(results.totalSize.width - Self.yLabelPadding,
-                                            results.plotBorderRect.midY - y2LabelSize.width/2)
+    /// Makes adjustments to the layout as requested by the plot.
+    private func adjustPlotSize(_ plotSize: Size, info: AdjustsPlotSize) -> Size {
+        if info.desiredPlotSize != .zero {
+            // Validate the requested size.
+            guard info.desiredPlotSize.height <= plotSize.height,
+                info.desiredPlotSize.width <= plotSize.width else {
+                    print("Size requested by plot is too large. Cannot fulfill: \(info.desiredPlotSize)")
+                    return plotSize
+            }
+            return info.desiredPlotSize
         }
+        return plotSize
     }
     
-    private func calcMarkerTextLocations(renderer: Renderer, results: inout Results) {
-        
-        for i in 0..<results.plotMarkers.xMarkers.count {
-            let textWidth = renderer.getTextWidth(text: results.plotMarkers.xMarkersText[i], textSize: markerTextSize)
-            let text_p = Point(results.plotMarkers.xMarkers[i] - (textWidth/2), -2.0 * markerTextSize)
-            results.xMarkersTextLocation.append(text_p)
-        }
-        
-        for i in 0..<results.plotMarkers.yMarkers.count {
-            var textWidth = renderer.getTextWidth(text: results.plotMarkers.yMarkersText[i], textSize: markerTextSize)
-            textWidth = min(textWidth, yMarkerMaxWidth)
-            let text_p = Point(-textWidth - 8, results.plotMarkers.yMarkers[i] - 4)
-            results.yMarkersTextLocation.append(text_p)
+    /// Rounds the given markers to integer pixel locations, for sharper gridlines.
+    private func roundMarkers(_ markers: inout PlotMarkers) {
+        //      for i in markers.xMarkers.indices {
+        //        markers.xMarkers[i].round(.down)
+        //      }
+        //      for i in markers.yMarkers.indices {
+        //        markers.yMarkers[i].round(.down)
+        //      }
+        //      for i in markers.y2Markers.indices {
+        //        markers.y2Markers[i].round(.down)
+        //      }
+    }
```

Comment:
Is this going to be tackled in a future PR?

Reply:
In order to get sharp gridlines we would need to have into account the thickness of the lines over here, something like this:
```swift
markers.xMarkers[i].round(.down)
if !Int(grid.thickness).isMultiple(of: 2) {
    markers.xMarkers[i] += 0.5
}
```

Reply:
Possibly.

And yes, you're correct - we actually need to align to a pixel boundary, not just an integer boundary. They might not always be the same. Also, it's kind of a controversial change because it means your markers will use a slightly different scale to your data. 

Perhaps I should just remove this.